src/bus/match.c

```diff
@@ -339,6 +339,62 @@ static bool match_keys_match_filter(MatchKeys *keys, MatchFilter *filter) {
         return true;
 }
 
+static int match_registry_by_path_compare(CRBTree *tree, void *k, CRBNode *rb) {
+        MatchRegistryByPath *registry = c_container_of(rb, MatchRegistryByPath, registry_node);
+        const char *path1 = k ?: "", *path2 = registry->path;
```

Comment:
Mh. This is hidden deep down. Shouldn't the caller of the lookup pick the empty string rather than NULL?

Reply:
As discussed, leaving this as is for now.